File: c/readability-score.c
#include <cs50.h>
#include <stdio.h>
#include <ctype.h>
#include <string.h>
#include <math.h>

int main(void)
{
    // Prompt user for text
    string text = get_string("Please type in text: ");

    // counts for letters, words, sentences
    float letters = 0;
    float words = 1;      // start with 1 to account for last word
    float sentences = 0;

    // Loop for counting
    for (int i = 0; i < strlen(text); i++)
    {
        // Counting letters, isalpha returns non 0 for alphabetical
        char c = text[i];

        if (isalpha(c) != 0)
        {
            letters++;
        }
        // Counting words, chars after spaces count as words
        if (c == ' ')
        {
            words++;
        }
        // Counting Sentences, any punctuation that ends a sentence
        if (c == '.' || c == '!' || c == '?')
        {
            sentences++;
        }
    }

    // Calculation, Coleman-Liau Index = 0.0588 * L - 0.296 * S - 15.8
    // per 100 words in text, L is avg num of letters, S is avg num of sentances

    float L = (letters / words) * 100;
    float S = (sentences / words) * 100;
    int index = round(0.0588 * L - 0.296 * S - 15.8);

    if (index > 16)
    {
        printf("Grade 16+\n");
    }
    else if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else
    {
        printf("Grade %i\n", index);
    }
}

File: python/readability-score.py
from cs50 import get_string

text = get_string("Please enter text: ")

# declare vars for letters, words (1 ie. words == " " + 1), sentences
letters = 0
words = 1
sentences = 0

for char in text:
    if char.isalpha():
        letters += 1
    elif char == " ":
        words += 1
    elif char == "." or char == "!" or char == "?":
        sentences += 1

# calculate Coleman-Liau index, 0.0588 * L - 0.296 * S - 15.8
# L is avg letters per 100 words, S is avg sentences per 100 words
avg100 = 100 / words
L = letters * avg100
S = sentences * avg100
index = round(0.0588 * L - 0.296 * S - 15.8)

# print grade level per index
if index < 1:
    print("Before Grade 1")
elif index >= 16:
    print("Grade 16+")
else:
    print(f"Grade {index}")
